Variable Substitution Relative Path

adstep opened the pull request:
adding 'relFile' to support substituting the path to the current open file realtive to cwd

Issue: https://github.com/Microsoft/vscode/issues/7000

msftclas commented:
Hi __@adstep__, I'm your friendly neighborhood Microsoft Pull Request Bot (You can call me MSBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.microsoft.com.
    </span>

TTYL, MSBOT;

msftclas commented:
__@adstep__, Thanks for signing the contribution license agreement so quickly! Actual humans will now validate the agreement and then evaluate the PR.
<br />Thanks, MSBOT;

adstep commented:
Related Issue: https://github.com/Microsoft/vscode/issues/7000

dbaeumer commented:
@adstep thanks for the PR. I have a couple of comments:

- the spec should be that the file path is relative to the open workspace folder. Otherwise someone could assume it is relative to process.cwd.
- we need to specify what the value would be if the uses opens a single file (there is no workspace folder and therefore no cwd)
- I would name it relativeFile. There is IMO no need for the abbreviation.

adstep commented:
@dbaeumer thanks for the comments!

* I've changed the code to use the workspaceRoot instead of cwd.
..* On a side note it looks like cwd is just an alias for workspaceRoot - "the task runner's current working directory on startup". Is this intended, or should cwd actually be process.cwd?
* I modified it so that if workspaceRoot is undefined the file path is returned. 
..* However, I'm unsure if this case will ever get hit because tasks are only available on a workspace folder.
* Renamed to relativeFile agreed

dbaeumer commented:
@dbaeumer: cwd being workspaceRoot is intended. I is named cwd since it sets the cwd of the spawn process. But I agree, confusing. Agree with task only being supported in workspace folder, however the systemVariables are used outside as well (e.g. debugging).

dbaeumer commented:
And thanks again for the nice PR.
